r/R/worker_helpers.R

```diff
@@ -143,11 +143,44 @@ getClusters <- function(type, resolution, data) {
   return(data)
 }
 
-handle_pagination <- function(gene_results, offset, limit, order_by, order_decreasing, filter) {
-  if (!is.null(filter)) {
-    gene_results <- gene_results[grepl(filter, gene_results$gene_name, ignore.case = TRUE), ]
+applyFilters <- function(gene_results, filters) {
+
+  filter_columns <- sapply(filters, `[[`, 'columnName')
+
+  # apply gene text filter
+  gene.idx <- which(filter_columns == 'gene_names')[1]
+  if (!is.na(gene.idx)) {
+    gene <- filters[[gene.idx]]$expression
+    gene_results <- gene_results[grepl(gene, gene_results$gene_names, ignore.case = TRUE), ]
   }
 
+  # apply numeric filters
+  numeric_columns <- c('logFC', 'p_val_adj', 'pct_1', 'pct_2', 'auc')
+
+  for (idx in seq_along(filter_columns)) {
+    column <- filter_columns[idx]
+    if (!column %in% numeric_columns) next()
+    filter <- filters[[idx]]
+
+    gene_results <- applyNumericFilter(gene_results, column, filter$comparison, filter$value)
+  }
+
+  return(gene_results)
+}
+
+applyNumericFilter <- function(gene_results, column, comparison, value) {
```

Comment:
no equal?

Reply:
>= / <=

Reply:
>= / <=

Reply:
Wont this lead to some border case? I know exact value will be very hard to hit but still. Something like >= / <=

Reply:
I think the motivation was to simplify things as e.g. logFC > 0 implies upregulated not logFC >= 0 and then everything else as you said hitting exact values won't happen so it's not necessary.